average.js

```diff
@@ -1,6 +1,13 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  // return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  
+  // the problem here is that many of the values in `numbers` array can be NaNs
+  // as expected from various real world sensors.
+  // So the solution is to filter out the NaNs and then calculate the average.
+  const realNumbers = numbers.filter(num => !isNaN(num));
+  const avg = realNumbers.reduce((p, c) => p+c, 0) / realNumbers.length;
+  return avg;
 }
 
 module.exports = {average};
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
Consider this situation: In IoT applications, analog sensors will report occasional outliers. 'Outliers' are values so small or so large, they aren't expected to happen. We need new functionality to ignore these outliers.
First add a unit-test in `test/average.test.js` to supply outliers and expect them to be ignored. Name the test appropriately. Then implement it in the code.
You'll need thresholds to recognize outliers: Think of sensing battery-temperature values and determine reasonable values.